modules/options/src/main/scala/scala/build/Artifacts.scala

```diff
@@ -415,24 +483,31 @@ object Artifacts {
     classifiersOpt: Option[Set[String]],
     maybeRecoverOnError: BuildException => Option[BuildException] = e => Some(e)
   ): Either[BuildException, Fetch.Result] = either {
+    val (_, res) = value {
+      fetchKeepFetcher(
+        dependencies,
+        extraRepositories,
+        paramsOpt,
+        logger,
+        cache,
+        classifiersOpt,
+        maybeRecoverOnError
+      )
+    }
+    res
+  }
+
+  private def fetchKeepFetcher(
```

Comment:
This name is a bit confusing, I think... some scaladocs would help, maybe 🤔 
It's hard to get around between `fetch`, `fetch0` and `fetchKeepFetcher`...